### DISCUSSION TITLE
Deferred deletion of root object ( in an object graph)

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#2976] [General]
We have never used deferred deletion on root objects, I was testing it today and I couldn't get it working.
The steps I followed.
1) Retrieved the object from the database.
2) Marked it for deferred deletion by calling something like obj = obj.Delete();
3) Saved the object obj = obj.Save();

It appears the object has to be 'dirty' before Save() would do anything?
FYI Immediate deletion has always worked for us.

Regards.

**Comment by TheCakeMonster** [comment]
Hmm, interesting. I would not have expected a problem. However, it's not a scenario in which I would normally use deferred deletion, I suppose. Normally I'm using it on items in a collection, because it's the collection as a whole - or the parent of it - that is being saved.

**Reply by Chicagoan2016** [reply]
It turned out to be our fault. We had written our own base classes (derived from Csla base classes) and there was a bug in our classes.
